addons/base_import/models/base_import.py

```diff
@@ -476,107 +593,239 @@ def _try_match_date_time(self, preview_values, options):
+            if not filtered_fields:
+                return {}
+
+            fields_matching_distance = {}
+            for field in filtered_fields:
+                field_string = field.get('string', '').casefold()
+
+                # use string distance for fuzzy match only on most likely field types
+                name_field_dist = self._get_distance(header.casefold(), field['name'].casefold())
+                string_field_dist = self._get_distance(header.casefold(), field_string)
+                translated_string_field_dist = self._get_distance(translated_header.casefold(), field_string)
+
+                fields_matching_distance[field['name']] = min([name_field_dist, string_field_dist, translated_string_field_dist])
+
+            # Keep only the closest mapping suggestion. Note that in case of multiple mapping on the same field,
+            # a mapping suggestion could be canceled by another one that has a smaller distance on the same field.
+            # See 'deduplicate_mapping_suggestions' method for more info.
+            min_dist_field = min(fields_matching_distance, key=fields_matching_distance.get)
+            distance = fields_matching_distance[min_dist_field]
+            if distance < self.FUZZY_MATCH_DISTANCE:
+                return {
+                    'field_path': [next(field['name'] for field in filtered_fields if field['name'] == min_dist_field)],
+                    'distance': distance
+                }
+            else:
+                return {}
 
         # relational field path
-        traversal = []
-        subfields = fields
+        field_path = []
+        subfields_tree = fields_tree
         # Iteratively dive into fields tree
-        for section in header.split('/'):
-            # Strip section in case spaces are added around '/' for
+        for sub_header in header.split('/'):
+            # Strip sub_header in case spaces are added around '/' for
             # readability of paths
-            match = self._match_header(section.strip(), subfields, options)
+            # Skip Saved mapping (mapping_field = {})
+            match = self._get_mapping_suggestion(sub_header.strip(), subfields_tree, header_types, {})
             # Any match failure, exit
             if not match:
-                return []
-            # prep subfields for next iteration within match[0]
-            field = match[0]
-            subfields = field['fields']
-            traversal.append(field)
-        return traversal
-
-    def _match_headers(self, rows, fields, options):
+                return {}
+            # prep subfields for next iteration within match['name'][0]
+            field_name = match['field_path'][0]
+            subfields_tree = next(item['fields'] for item in subfields_tree if item['name'] == field_name)
+            field_path.append(field_name)
+        # No need to return distance for hierarchy mapping
+        return {'field_path': field_path}
+
+    def _get_distance(self, a, b):
```

Comment:
Could copy the comment in method that states that "0 = exact match and 1 = completely different strings" to give bit of context as to what is returned here.